src/sentry/api/endpoints/organization_transaction_details.py

```diff
@@ -0,0 +1,363 @@
+                    "op": segment_span.get("op"),
+                    "status": SPAN_STATUS_CODE_TO_NAME[segment_span.get("status", 2)],
+                    "exclusive_time": segment_span.get("exclusive_time"),
+                    "hash": segment_span.get("group"),
+                    "type": "trace",
+                },
+            },
+            "culprit": transaction_name,
+            "environment": segment_sentry_tags.get("environment"),
+            "errors": [],
+            "extra": {},
+            "ingest_path": [],
+            "level": "info",
+            "location": transaction_name,
+            "logger": "",
+            "measurements": segment_measurements,
+            "metadata": {
+                "location": transaction_name,
+                "title": transaction_name,
+            },
+            "nodestore_insert": segment_span.get("precise.finish_ts"),
+            "received": segment_span.get("precise.finish_ts"),
+            "request": {
+                # missing from indexed spans dataset
+            },
+            "sdk": {
+                "name": segment_sentry_tags.get("sdk.name"),
+                "version": segment_sentry_tags.get("sdk.version"),
+                # other props missing from indexed spans dataset
+            },
+            "span_grouping_config": {
+                # missing from indexed spans dataset
+            },
+            "entries": [
+                {
+                    "type": "spans",
+                    "data": _span_data_to_event_spans(spans_data),
+                }
+            ],
+            "start_timestamp": segment_span.get("precise.start_ts"),
+            "timestamp": segment_span.get("precise.finish_ts"),
+            "title": transaction_name,
+            "transaction": transaction_name,
+            "transaction_info": {
+                # missing from indexed spans dataset
+            },
+            "type": "transaction",
+            "user": {
+                # missing from indexed spans dataset
+            },
+            "version": 5,
+            "projectSlug": project_slug,
+        }
+
+        average_columns = request.GET.getlist("averageColumn", [])
+        if (
+            all(col in VALID_AVERAGE_COLUMNS for col in average_columns)
+            and len(average_columns) > 0
+        ):
+            _add_comparison_to_event(data, organization.id, project, average_columns)
+
+        return Response(data)
+
+
+def _query_all_spans_in_transaction(organization, project, transaction_id, start, end):
+    params: ParamsType = {
+        "start": start - timedelta(seconds=1),
+        "end": end + timedelta(seconds=1),
+        "project_id": [project.id],
+        "project_objects": [project],
+        "organization_id": organization.id,
+    }
+
+    # Look up the spans for this transaction
+    query = SpansIndexedQueryBuilder(
+        Dataset.SpansIndexed,
+        params,
+        query=f"transaction_id:{transaction_id}",
```

Comment:
🤔 Do we have a way to get `trace_id` here before we query the spans table? I know we don't have one on txn_id yet, but I added an index on trace_id 2 weeks ago so it might help the perf somewhat at least

Reply:
Could we leave this to the frontend to pass? (Similar to the start/end timestamps?) The FE should already know the trace ID most of/all the time